src/mmrelay/matrix/command_bridge.py

```
@@ -23,6 +27,283 @@
+                target = f"@{target}"
+        target = target.split("?", maxsplit=1)[0]
+        return target if target.startswith("@") else None
+
+    if decoded_href.startswith("@"):
+        return decoded_href.split("?", maxsplit=1)[0]
+
+    return None
+
+
+def _normalize_formatted_body_for_command_detection(
+    formatted_body: Any, *, bot_mxid: str | None
```

Ruff (0.15.10)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `formatted_body`

(ANN401)